format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -38,10 +40,20 @@
    * @param dataTypeDetectorStatusKeeper Object that keeps the state of automated data type detection process.
    * @param columnNames Column names.
    * @param rowValues Row values.
+   * @param collector Failure Collector to store failures
    */
   public static void detectDataTypeOfRowValues(Map<String, Schema> override,
                                                DataTypeDetectorStatusKeeper dataTypeDetectorStatusKeeper,
-                                               String[] columnNames, String[] rowValues) {
+                                               String[] columnNames, String[] rowValues,
+                                               @Nullable FailureCollector collector) {
+    if (rowValues.length < columnNames.length) {
+      if (collector != null) {
+        collector.addFailure(String.format(
+          "Number of values in row (%d) is less than number of values in header (%d).",
```

Comment:
Its possible that `header` is not present in csv file. Should we change error message to `number of columns`?

Reply:
You are right, changed message.